portfolio/src/main/webapp/script.js

```diff
@@ -26,3 +27,55 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+
+
+function randomizeImage() {
+  // The images directory contains 13 images, so generate a random index between
+  // 1 and 13.
+  const imageIndex = Math.floor(Math.random() * 7) + 1;
+  const imgUrl = 'images/IMG_' + imageIndex + '.jpeg';
+
+  const imgElement = document.createElement('img');
+  imgElement.src = imgUrl; 
+
+  var imageContainer = document.getElementById('random-image-container');
+  // Remove the previous image.
+  imageContainer.innerHTML = '';
+  imageContainer.appendChild(imgElement);
+}
+
+function showFrame() {
+    document.getElementById("showiFrame").innerHTML = "<iframe src=\"https://open.spotify.com/embed/track/1bSpwPhAxZwlR2enJJsv7U\" width=\"300\" height=\"380\" frameborder=\"0\" allowtransparency=\"true\" allow=\"encrypted-media\" ></iframe>";
+
+
+}
+
+function getDataUsingArrowFunctions() {
+  fetch('/data').then(response => response.text()).then((quote) => {
+    document.getElementById('data-container').innerText = quote;
+  });
+}
+
+/*function getData() {
+    fetch('/data').then(response => response.json()).then((data) => {
+    // stats is an object, not a string, so we have to
+    // reference its fields to create HTML content
+
+    const dataElements = document.getElementById('data-container');
+    statsListElement.innerHTML = '';
+    statsListElement.appendChild(
+        createListElement('Element 1: ' + data.get(0)));
+    statsListElement.appendChild(
+        createListElement('Element 2: ' + data.get(1)));
+    statsListElement.appendChild(
+        createListElement('Element 3: ' + data.get(2)));
+  });
+}*/
```

Comment:
Nit: Probably want to clean up commented out functions and lines, e.g. here and a few lines your doPost function.